businessRules/users.js

```diff
@@ -0,0 +1,49 @@
+const { v4 } = require('uuid');
+const md5 = require('md5');
+const { createUser } = require('../models/users');
+const { getUsers, setUsers } = require('../utils/storage');
+
+function registerUser(userName, password, firstName, lastName) {
+    firstName = firstName || 'Anonymous';
+    lastName = lastName || 'Incognito';
+
+    createUser(userName, password, firstName, lastName);
+}
+
+function login(req, res) {
+    let users = getUsers();
+
+    const userName = req.body.userName;
+    const password = md5(req.body.password);
+
+    const user = users.find(
+        (user) => user.userName === userName && user.password === password
+    );
+    if (!user) {
+        res.status(400).json({ status: 'No such user.' });
+        return;
+    }
+
+    if (user) {
```

Comment:
This check is not needed if you have previously checked for `!user`.

Reply:
Updated and tested. )